File: codewar/V1/Day-145/bitCounting.js
// https://www.codewars.com/kata/526571aae218b8ee490006f4/train/javascript

/**
 *
 * @param {number} n
 */
const countBits = (n) => {
  // const nBit = n.toString(2);
  // console.log(nBit);
  // return nBit.split("").filter((item) => {
  //   return Number(item === "1");
  // }).length;
  return n
    .toString(2)
    .split("")
    .filter((item) => Number(item === "1")).length;
};

const test = 123;
// console.log(test.toString(2));
// console.log((test >>> 0).toString(2));

// console.log(countBits(test));

/* ----------------------------- other solution ----------------------------- */
/**
 *
 * @param {Number} n
 */
function countBits2(n) {
  return n.toString(2).split("0").join("").length;
}

File: codewar/V1/Day-145/bitCounting.py
# https://www.codewars.com/kata/526571aae218b8ee490006f4/train/python
def count_bits(n):
    """find the bits and only count 1

    Args:
        n (number): number
    """
    return len(bin(n).replace("0b", "").replace("0", ""))
    # or
    # return bin(n).count('1')


print(count_bits(22))


def first(n):
    # or this
    count = 0
    while n > 0:
        if n & 1:
            count += 1
        n >>= 1
    return count


print(first(22))
